packages/cli/src/ceramic-daemon.ts

```diff
@@ -683,13 +712,14 @@ export class CeramicDaemon {
       code: result.payload.code,
       requestPath: result.payload.requestPath,
       models: result.payload.requestBody ? result.payload.requestBody.models : undefined,
+      modelData: result.payload.requestBody ? result.payload.requestBody.modelData : undefined,
     }
   }
 
   private async _validateAdminApiJWS(
     basePath: string,
     jws: string | undefined,
-    shouldContainModels: boolean
+    successCallback?: AdminApiMutationMethod
```

Comment:
does this actually need the full successCallback?  Can't it just take the 'type' string?  I feel like it makes this method confusing and obfuscates its core purpose of validation by also having it pass around the successCallback

Reply:
We need a way to know what arguments to pass to each method. Otherwise we end up back where we started, where the indexing methods have to validate that modelIds or modelData are present. Although you would expect the error to not occur, it would be a little weird to return an error to the user that is "JWS Validation passed back model id's when I was expecting modelData"

Reply:
don't we already know in the caller which argument we need to use because of the context from which api method was called?  In `startIndexingModelData` we can just always take the `modelData` field from the input and ignore `modelIds`.  If we passed the validation from this function (`_validateAdminApiJWS`), then we know the data will be present

Reply:
Yeah, if we want to assume that the fields will be present after passing validation, then we don't have to use the callback approach. I was mainly using that in case the code changes, so that the code will break at compile time rather than run time.

Reply:
I don't quite follow how this structure helps catch more errors at compile time?  What kinds of issues would this catch that separating validation from execution would not?  Could these be solved by having validation return a more restricted type with just the parsed and validated fields back to the caller to then be passed into the execution layer?

I wonder how much of this will all become moot once we use a codec here

Reply:
for the record, I don't feel that strongly about this thread of comments - this is internal code structure we can always clean up later.  My other comment thread about the actual API structure is the more important comment and the only one I would hold up merging this PR over.